### PR TITLE
feat: remove tools.mk dependency

### PR DESCRIPTION
**Pull request by viktorvoltaire**
Since we generate makefiles. there's no real reason to keep `tools.mk` around

### CODE REVIEW

**Review by odsod — APPROVED** [review]
Bloody brilliant! :sparkler:

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,15 @@
...
+
+$(magefile): $(mage_dir)/go.mod $(shell find $(mage_dir)/.. -type f -name '*.go')
+	@cd $(mage_dir) && go run go.einride.tech/mage-tools gen
+
+.PHONY: mage-clean
+mage-clean:
+	@git clean -fdx $(mage_dir)
 
 .PHONY: all
 all: $(mage)
```
Comment (odsod) [inline Makefile:16]: the `$(mage)` here should also get updated?

Comment (viktorvoltaire) [reply Makefile]: Fixed

**Review by odsod — APPROVED** [review]
Let's go :raised_hands:

**Pull request merged**